libcontainer/process_linux.go

```diff
@@ -232,7 +255,11 @@ func (p *setnsProcess) start() (retErr error) {
 		return fmt.Errorf("error executing setns process: %w", err)
 	}
 	for _, path := range p.cgroupPaths {
-		if err := cgroups.WriteCgroupProc(path, p.pid()); err != nil && !p.rootlessCgroups {
+		procPid := p.pid()
+		if useClone3 {
+			procPid = -1
+		}
+		if err := cgroups.WriteCgroupProc(path, procPid); err != nil && !p.rootlessCgroups {
```

Comment:
I guess if it's -1 it's not written? Let's just useClone3 for the condition, explaining that if we are using clone3, then it's already set in the cgroup.